sql/log_event.h

```diff
@@ -2706,11 +2804,15 @@ class Unknown_log_event: public binary_log::Unknown_event , public Log_event
   Unknown_log_event(const char* buf,
                     const Format_description_event *description_event)
   : binary_log::Unknown_event(buf, description_event),
-    Log_event(header(), footer())
+    Log_event(header(), footer()), what(UNKNOWN)
   {
     is_valid_param= true;
   }
 
+  /* constructor for hopelessly corrupted events */
```

Comment:
Please add a comment why we assume that corrupted event is encrypted

Reply:
Added